src/js/components/Person.js

```diff
@@ -107,43 +102,80 @@ export default class Person extends Component {
+
+    if (timezone) {
+      return `It is ${currentHour}${ampm} in ${city} (${timezone} UTC).`;
+    } else {
+      return `It is ${currentHour}${ampm} in ${city}.`;
     }
+  }
+
+  _getHourFromLDAPTimezone (personTimezone) {
+    const currentDate = new Date();
+    // converting minutes to milliseconds for localOffset
+    const localOffset = currentDate.getTimezoneOffset() * 60000;
+    const localTime = currentDate.getTime();
+    const localUTC = localTime + localOffset;
+    // expect personTimezone to look like "+0100"
+    // take positive or negative sign, and convert to int
+    const offsetSign = Number.parseInt(personTimezone.substr(0, 1) + '1');
+    // taking hours offset
+    const personHoursOffset = Number.parseInt(personTimezone.substr(1, 2));
+    // taking last two "digits" from string to convert to decimal
+    const personMinutesOffset = Number.parseInt(personTimezone.substr(-2)) / 60;
+    const personTimezoneOffset = (personHoursOffset + personMinutesOffset) * offsetSign;
+    // converting personTimezoneOffset from hours to milliseconds
+    const personDate = new Date(localUTC + (3600000 * personTimezoneOffset));
 
-    return <Status value={value} />;
+    return personDate.getHours();
   }
 
-  _getTimezone (latitude, longitude) {
+  _getTimezone (location) {
     const person = this.state.person;
-    const params = {
-      lat: latitude,
-      lng: longitude,
-      username: GEONAMES_USERNAME
-    };
+    let currentPersonTime;
 
-    Rest
-      .get("http://api.geonames.org/timezoneJSON", params)
-      .end((err, res) => {
-        if (err) {
-          this.setState({currentPersonTime: '', error: err});
-        } else if (res.ok) {
-          const result = res.body;
-          const time = result.time;
-
-          let personHour = Number.parseInt(time.substr(11, 2));
-          const ampm = personHour >= 12 ? 'pm' : 'am';
-          personHour = personHour % 12;
-          // account for midnight
-          personHour = personHour ? personHour : 12;
-          const currentPersonTime = `It is ${personHour}${ampm} in ${person.l}.`;
-
-          this.setState({currentPersonTime: currentPersonTime, error: null});
-        }
-      });
+    if (location.latitude && location.longitude) {
+      const params = {
+        lat: location.latitude,
+        lng: location.longitude,
+        username: GEONAMES_USERNAME
+      };
+
+      Rest
+        .get("http://api.geonames.org/timezoneJSON", params)
+        .end((err, res) => {
+          if (err) {
+            this.setState({currentPersonTime: '', error: err});
+          } else if (res.ok) {
+            const result = res.body;
+            const time = result.time;
+            const personHour = Number.parseInt(time.substr(11, 2));
+            currentPersonTime = this._formatHourInCity(personHour, person.l);
+
+            this.setState({currentPersonTime: currentPersonTime, error: null});
+          }
+        });
+    } else if (location.timeZone) {
```

Comment:
Updated the PR so the setState happens only once in the else block instead of the ldap timeZone fallback if block.